drivers/sensor/lsm6dsl/lsm6dsl.c

```diff
@@ -476,9 +476,9 @@ static inline void lsm6dsl_accel_convert(struct sensor_value *val, int raw_val,
 
 	/* Sensitivity is exposed in mg/LSB */
 	/* Convert to m/s^2 */
-	dval = (double)(raw_val) * (double)sensitivity * SENSOR_G_DOUBLE / 1000;
-	val->val1 = (int32_t)dval;
-	val->val2 = (((int32_t)(dval * 1000)) % 1000) * 1000;
+	dval = (double)(raw_val) * (double)sensitivity / 1000; 
+	int32_t dval_ug = dval * 1000000; //value converted in micro g
```

Comment:
Sorry if I noticed only now (my bad), but I'm not sure it is allowed to declare a variable in the middle of a body function. I don't remember any specific rule, but I never noticed it. Maybe @fabiobaltieri or @MaureenHelm may confirm it or not.

Reply:
Moreover, if you have already a  value espressed in g, maybe you can directly use the sensor_g_to_ms2.
Alternatively we can change the sensitivity in .h and express it as 61 ug/LSB (you can take a look to #71223 that I just wrote looking at this work. I left lsm6dsl out to let you finish it)

Reply:
@avisconti Sorry i missed your reply.

I don't understand your comment about the variable in the middle of a body function, do you mean that the variable should be declared at the top, like dval, or that everything should be done without using additional variables?

I'll look at your pull request and add the same modifications to lsm6dsl

Reply:
I can't recall a specific guideline, it's generally discouraged depending on the reviewer (I would flag it) unless it really improves the code readability.

Blocks or declare and initialize at the top are commonly used (all the cfg/data pairs for example), in this case I'd say either declare and initialize both `dval` and `dval_ug` or declare and initialize separately both, keep the styling coherent with the function and ideally with the whole file.

Reply:
Also the comment format is off, but also the comment are redundant, the variable name already includes the unit (as it should be), drop the comment